crates/turbopack-dev-server/src/source/mod.rs

```diff
@@ -91,6 +93,14 @@ pub enum ContentSourceContent {
     Rewrite(RewriteVc),
 }
 
+/// This trait can be emitted as collectible and will be applied after the
+/// request is handled and it's ensured that it finishes before the next request
+/// is handled.
```

Comment:
Can you explain why you would want to do emit side effect vcs? What does it accomplish?

Reply:
It moves the invalidation to a time after the request is sent to the user. This delays all computation for the state change to some time after this request and before the next request.

If we would just change it inline, the strongly consistent read from the request handling would wait for this invalidation to be completed as it might have effects on the current request (but it doesn't have these).

So for expensive invalidations this allows to respond ealier.

Reply:
Technically it's also possible to return a list of SideEffectVcs, but this requires adding a list to all intermediate functions, which is quite a large change. Since we have to cool system of collectibles we can use that. It allows to emit a side effect from anywhere and the server will make sure that it's handled at the right time.